applications/vanilla/controllers/api/ThemesApiController.php

```diff
@@ -0,0 +1,394 @@
+            "header?" => Schema::parse([
+                "type:s" => ['description' => 'Header asset type: html.', 'enum' => ['html']],
+                "body:s"
+            ]),
+            "footer?" => Schema::parse([
+                "type:s" => ['description' => 'Footer asset type: html.', 'enum' => ['html']],
+                "body:s"
+            ]),
+            "variables?" => Schema::parse([
+                "type:s" => ['description' => 'Variables asset type: json.', 'enum' => ['json']],
+                "data:o"
+            ]),
+            "fonts?" => Schema::parse([
+                "type:s" => ['description' => 'Fonts asset type: data.', 'enum' => ['data']],
+                'data:a' => Schema::parse([
+                    "type:s",
+                    "name:s",
+                    "fallbacks:s?",
+                    "url:s"
+                ])
+            ]),
+            "scripts?" => Schema::parse([
+                "type:s" => ['description' => 'Scripts asset type: data.', 'enum' => ['data']],
+                "data:a"
+            ]),
+            "styles?" => Schema::parse([
+                "type:s" => ['description' => 'Styles asset type: css.', 'enum' => ['css']],
+                "data:s"
+            ]),
+            "inlineJs?" => Schema::parse([
+                "type:s"=> ['description' => 'Javascript asset type: js.', 'enum' => ['js']],
+                "js:s"
+            ])
+        ]);
+        return $schema;
+    }
+
+    /**
+     * Get custom theme by ID
+     *
+     * @param int $themeID
+     * @return array
+     */
+    public function getThemeByID(int $themeID) {
+        $theme = $this->themeModel->get(['themeID' => $themeID]);
+        if (empty($theme)) {
+            throw new NotFoundException('Theme '.$themeID.' not found');
+        }
+        return $theme[0];
+    }
+
+    /**
+     * Get theme by name
+     *
+     * @param string $themeName
+     * @return Addon Returns theme addon
+     *
+     * @throws NotFoundException Throws an exception when themeName not found.
+     */
+    public function getThemeByName(string $themeName): Addon {
+        $theme = $this->addonManager->lookupTheme($themeName);
+        if (null === $theme) {
+            throw new NotFoundException('There is no theme: \''.$themeName.'\' installed.');
+        }
+        return $theme;
+    }
+
+    /**
+     * Get list of all available themes
+     *
+     * @return array List of all available themes
+     */
+    public function getAllThemes(): array {
+        $themes = $this->addonManager->lookupAllByType(Addon::TYPE_THEME);
+        $result = [];
+        foreach ($themes as $theme) {
+            $result[] = $theme->getKey();
+        }
+        return $result;
+    }
+
+    /**
+     * Get all theme assets
+     *
+     * @param string $themeKey
+     * @return array
+     */
+    public function getThemeAssets(string $themeKey): array {
+        $customAssets = [];
+        if (ctype_digit($themeKey)) {
```

Comment:
I think this function could be improved by pulling this piece of logic into some `getThemeByKey` with a docblock explaining the how the discimination between the id and names are made.